src/main/java/com/arpnetworking/clusteraggregator/aggregation/AggMessageExtractor.java

```diff
@@ -52,6 +53,18 @@ public String entityId(final Object message) {
                     .append(metricData.getMetric())
                     .append("||")
                     .append(metricData.getPeriod());
+            for (final Messages.DimensionEntry dimensionEntry : metricData.getDimensionsList()) {
+                final String k = dimensionEntry.getKey();
+                if (!(k.equals(AggClientConnection.CLUSTER_KEY)
```

Comment:
Yeah I have a branch where I de-duped and sorted them - what are your thoughts on de-duplication? Can we assume there are no duplicate keys from MAD?